### PR TITLE
[FIX] Doxygen warnings in Core Module

### PR DESCRIPTION
**Pull request by joergi-w**
This PR improves the SeqAn documentation in the core module and removes all 61 warnings from the doxygen build.

### CODE REVIEW

File: include/seqan3/core/tuple_utility.hpp [hunk 1]
```diff
@@ -127,7 +129,6 @@ constexpr auto tuple_split(tuple_t<ts...> && t)
  * \ingroup core
  *
  * \tparam    pivot_t A template type specifying the split position.
- * \param[in] t       The original tuple to split.
```
[the remark this answers is not in the document]
Comment (smehringer) [reply include/seqan3/core/tuple_utility.hpp]: is this error reported in doxygen?

[the remark this answers is not in the document]
Comment (joergi-w) [reply include/seqan3/core/tuple_utility.hpp]: According to the documentation \copydetails copies everything except \brief, so this behaviour seems intended. I just copy the description manually here.

**Comment by joergi-w** [comment]
As discussed I reverted the namespace nesting. This PR is ready for a final review.

File: include/seqan3/core/pod_tuple.hpp [hunk 2]
```diff
@@ -63,34 +63,42 @@ struct pod_tuple<type0, types...>
     pod_tuple<types...> _tail;
     //!\endcond
 
-    //!\name Comparison operators
-    //!\{
-    //!\brief Lexicographically compares the values in the tuple.
+    /*!\name Comparison operators
+     * \{
+     * \brief Lexicographically compares the values in the tuple.
```
Comment (smehringer) [inline include/seqan3/core/pod_tuple.hpp]: I guess we do not need this if they all have the their own brief.

Comment (joergi-w) [reply include/seqan3/core/pod_tuple.hpp]: I agree!

**Pull request merged**